src/common/args.cpp

```diff
@@ -867,27 +867,15 @@ void ArgsManager::LogArgs() const
 
 namespace common {
 #ifdef WIN32
-WinCmdLineArgs::WinCmdLineArgs()
+WindowsScopedCodePage::WindowsScopedCodePage(UINT new_console_outp_code_page)
+    : m_original_code_page{GetConsoleOutputCP()}
 {
-    wchar_t** wargv = CommandLineToArgvW(GetCommandLineW(), &argc);
-    std::wstring_convert<std::codecvt_utf8_utf16<wchar_t>, wchar_t> utf8_cvt;
-    argv = new char*[argc];
-    args.resize(argc);
-    for (int i = 0; i < argc; i++) {
-        args[i] = utf8_cvt.to_bytes(wargv[i]);
-        argv[i] = &*args[i].begin();
-    }
-    LocalFree(wargv);
-}
-
-WinCmdLineArgs::~WinCmdLineArgs()
-{
-    delete[] argv;
+    SetConsoleOutputCP(new_console_outp_code_page);
 }
 
-std::pair<int, char**> WinCmdLineArgs::get()
+WindowsScopedCodePage::~WindowsScopedCodePage()
 {
-    return std::make_pair(argc, argv);
+    SetConsoleOutputCP(m_original_code_page);
```

Comment:
Is this necessary with the manifest?

Also we already have
```
    SetConsoleCP(CP_UTF8);
    SetConsoleOutputCP(CP_UTF8);
```
in `void SetupEnvironment()`.

Reply:
> Is this necessary with the manifest?

The manifest is responsible for the active code page, which differs from console input and output code pages. It would be easy to compare outputs of `GetACP()`, `GetConsoleCP()` and `GetConsoleOutputCP()`.

> Also we already have
> 
> ```
>     SetConsoleCP(CP_UTF8);
>     SetConsoleOutputCP(CP_UTF8);
> ```
> 
> in `void SetupEnvironment()`.

Thanks! I overlooked those calls. However, with the manifest, setting only the console output code page is sufficient.

>  It's not necessary to scope this as we want it for the entire run of the application.

The scoped implementation is necessary to restore the console code pages to their state prior to the application's execution.

Reply:
> The manifest is responsible for the active code page, which differs from console input and output code pages. It would be easy to compare outputs of GetACP(), GetConsoleCP() and GetConsoleOutputCP().

Thanks.

> Thanks! I overlooked those calls. However, with the manifest, setting only the console output code page is sufficient.

OK.

> The scoped implementation is necessary to restore the console code pages to their state prior to the application's execution.

We haven't been doing this ever. I think it was nice to simply roll this into `SetupEnvironment`. But okay.

Reply:
If we really need to, what if we make `SetupEnvironment` scoped instead? It's a more general solution, it doesn't belong in "common/args", and would work also if there's future environment modifications that need reverting before exit.